python/mlc_chat/nn/kv_cache.py

```diff
@@ -73,46 +84,119 @@ def create(  # pylint: disable=too-many-arguments
+            rx.extern("paged_kv_cache.attention_kernel_prefill_end_forward"),
+            rx.extern("paged_kv_cache.attention_kernel_decode_begin_forward"),
+            rx.extern("paged_kv_cache.attention_kernel_decode_end_forward"),
+            rx.extern("flashinfer.batch_qk_apply_rotary_in_place"),
+            rx.extern("flashinfer.merge_state_in_place"),
+            bb.add_func(
+                _kv_cache_debug_get_kv(num_hidden_layers, num_key_value_heads, head_dim, dtype),
+                "kv_cache_debug_get_kv",
+            ),
+        ]
+        super().__init__(
             _expr=rx.Call(
                 rx.extern("vm.builtin.paged_attention_kv_cache_create"),
-                args=[
-                    rx.ShapeExpr([max_batch_size, max_total_seq_len, page_size]),
-                    rx.PrimValue(num_hidden_layers),
-                    rx.PrimValue(num_attention_heads),
-                    rx.PrimValue(num_key_value_heads),
-                    rx.PrimValue(head_dim),
-                    rx.PrimValue(rope_scale),
-                    rx.PrimValue(rope_theta),
-                    rx.op.zeros((), dtype),
-                    bb.add_func(
-                        kv_cache_transpose_append(num_key_value_heads, head_dim, dtype),
-                        "kv_cache_transpose_append",
-                    ),
-                    rx.extern("paged_kv_cache.attention_kernel_prefill"),
-                    rx.extern("paged_kv_cache.attention_kernel_decode"),
-                    rx.extern("flashinfer.attention_kernel_prefill_with_ragged_kv_cache"),
-                    rx.extern(
-                        "flashinfer.attention_kernel_prefill_with_ragged_kv_cache_begin_forward"
-                    ),
-                    rx.extern(
-                        "flashinfer.attention_kernel_prefill_with_ragged_kv_cache_end_forward"
-                    ),
-                    rx.extern("paged_kv_cache.attention_kernel_prefill_begin_forward"),
-                    rx.extern("paged_kv_cache.attention_kernel_prefill_end_forward"),
-                    rx.extern("paged_kv_cache.attention_kernel_decode_begin_forward"),
-                    rx.extern("paged_kv_cache.attention_kernel_decode_end_forward"),
-                    rx.extern("flashinfer.batch_qk_apply_rotary_in_place"),
-                    rx.extern("flashinfer.merge_state_in_place"),
-                    bb.add_func(
-                        kv_cache_debug_get_kv(
-                            num_hidden_layers, num_key_value_heads, head_dim, dtype
-                        ),
-                        "kv_cache_debug_get_kv",
-                    ),
-                ],
+                args=args,
                 sinfo_args=[rx.ObjectStructInfo()],
             ),
             _name=name,
         )
+
+
+# mypy: disable-error-code="attr-defined"
+# pylint: disable=too-many-locals
+
+
+def _kv_cache_transpose_append(num_key_value_heads, head_dim, dtype):
```

Comment:
I moved the TIRs to this file, because:
- They are not operators used in this project
- They are closely attached to our paged KV cache